pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -168,15 +179,31 @@ func NewFlowExporter(connStore *connections.ConntrackConnectionStore, denyConnSt
 		ipfixSet:               ipfixentities.NewSet(false),
 		k8sClient:              k8sClient,
 		nodeRouteController:    nodeRouteController,
-		isNetworkPolicyOnly:    isNetworkPolicyOnly,
+		isNetworkPolicyOnly:    trafficEncapMode.IsNetworkPolicyOnly(),
 		nodeName:               nodeName,
-		conntrackPriorityQueue: conntrackPriorityQueue,
-		denyPriorityQueue:      denyPriorityQueue,
+		conntrackPriorityQueue: conntrackConnStore.GetPriorityQueue(),
+		denyPriorityQueue:      denyConnStore.GetPriorityQueue(),
 		expiredConns:           make([]flowexporter.Connection, 0, maxConnsToExport*2),
 	}, nil
 }
 
-func (exp *flowExporter) Run(stopCh <-chan struct{}) {
+func (exp *FlowExporter) GetDenyConnStore() *connections.DenyConnectionStore {
+	return exp.denyConnStore
+}
+
+func (exp *FlowExporter) GetConntrackConnStore() *connections.ConntrackConnectionStore {
+	return exp.conntrackConnStore
+}
+
+func (exp *FlowExporter) Run(stopCh <-chan struct{}) {
+	// Start the go routines to periodically delete stale deny connections.
+	denyConnStore := exp.GetDenyConnStore()
+	go denyConnStore.RunPeriodicDeletion(stopCh)
+
+	// Start the go routine to poll conntrack flows.
+	conntrackConnStore := exp.GetConntrackConnStore()
+	go conntrackConnStore.Run(stopCh)
```

Comment:
s/go routines/goroutines

I think you can just use `go exp.denyConnStore.Run()` and `go exp.conntrackConnStore.Run()`, without calling the getters

Reply:
That's right. I forgot to remove the getters. Thanks. Updated